datastream/datastream.py

```diff
@@ -440,7 +448,12 @@ def merge(datastreams_and_ns):
         )
 
     @staticmethod
-    def zip(datastreams):
+    def zip(datastreams: List[Datastream]) -> Datastream:
+        '''
+        Zip multiple datastreams together so that all combinations of examples
+        are possible creating tuples like ``(example1, example2, ...)``.
+        The samples are drawn independently from each underlying datastream.
```

Comment:
clarify if with or without replacement

Reply:
I don't think that belongs here but we should mention that the standard sample is using without replacement elsewhere

Reply:
Added some things to the explanation of datastream